### DISCUSSION TITLE
Support self-closing `x-slot` tag

### DISCUSSION THREAD

**Discussion by jacksleight** [laravel/framework#43765] [Ideas]
When you're using `x-slot` to just pass attributes to a child element (with no content) it would be great if you could just do:

```blade
<x-slot:button class="large" />
```

Unfortunately this doesn't work and errors with: `Unable to locate a class or view for component [slot:button].` You have to write out the full tag instead:

```blade
<x-slot:button class="large"></x-slot:button>
```

Only a small thing but it would be a nice way to make the code a bit cleaner.

**Comment by mikeydevelops** [comment]
Added a pull request #43879 for ya.

**Reply by jacksleight** [reply]
Amazing thanks! 😊 Sadly it was closed. 😕

**Reply by mikeydevelops** [reply]
> Amazing thanks! 😊 Sadly it was closed. 😕

Yeah, I'm sorry too, didn't see any reason tho.

**Reply by jacksleight** [reply]
Never mind, I guess the closing tags aren't really a problem, it'd just make things a bit tidier. Thanks for trying!
